report/report.tex

```diff
@@ -52,7 +52,11 @@ \subsection{Methods}
 
 For problem F we've solved each instance with Dijkstra's algorithm to find the path containing the fewest amount of red vertices. We've achieved this by assigning each edge entering a red vertex a weight of 1. Every other edge is assigned a vertex of 0. Thereby the path with the smallest cost to get from $s$ to $t$ must be the one that enters the fewest red vertices. Since Dijkstra only allows simple paths, the cost of the path must also be the total amount of red vertices that the path enters. Therefore the cost to get to $t$ is the total red vertices the path enters. In case there is no path from $s$ to $t$ the cost to get to $t$ will be $sys.maxsize$.
 
-For problem M...
+For problem M, we first check that the graph is directed and contains no cycles. This is because the problem can’t be solved if there is a cycle. If the graph is directed and with no cycles we compute a topological sorting starting from the start node. That is, any nodes pointing to the start node will not
+ be part of the topological sorting. We do this with DFS by including a boolean “is\_back” tag in the stack we pop from. If the tag on the current node is true we know that all nodes that can be reached from the current node have been visited, and thus we can add the node to our topological sorting. The result will thus be a reverse topological sorting, 
+such that we can pop from the sorting in order to obtain the first node in the topological sorting. When returning from the topological sorting, we also check whether the target node is in visited, since then we know that there exists a path from start to target. If this is not the case we can safely return
+ -1. If there is a path then we can go along and run our longest chain algorithm in order to find the path containing the most red vertices: We pop from the topological sorting and define ‘a’ as 1 if this node is red, otherwise 0. We then visit all neighbours and update their entry in the dist array: if the distance to our current node + a is bigger
```

Comment:
This should say: "If there is a path going from start to end vertex, we can run the longest chain algorithm in order to find the path containing the most red vertices"